packages/openmrs/src/Adaptor.js

```diff
@@ -317,15 +338,37 @@ export function getEncounters(query, callback = s => s) {
     const [resolvedQuery] = expandReferences(state, query);
     console.log('Fetching encounters by query', resolvedQuery);
 
-    const response = await request(
-      state,
-      'GET',
-      `/ws/rest/v1/encounter/`,
-      {},
-      resolvedQuery
-    );
+    let response;
+    let requestQuery = resolvedQuery;
+    let allowPagination = isNaN(requestQuery.startIndex);
+
+    do {
+      const res = await request(
+        state,
+        'GET',
+        `/ws/rest/v1/encounter/`,
+        {},
+        requestQuery
+      );
+
+      response
```

Comment:
Oh, you know what, I'd also like to see a log line for every page of data. Maybe something like:
```
fetching data...
fetched 20 results
```

Which repeats for every page

Reply:
Thanks @josephjclark i will work on the change request on Monday